### PR TITLE
Support `cfg` and `cfg_attr` on generic parameters

### PR DESCRIPTION
**Pull request by petrochenkov**
`cfg` attributes are supported in all other positions where attributes are accepted at all.

They were previously prohibited in https://github.com/rust-lang/rust/pull/51283 because they weren't implemented correctly before that and were simply ignored.

### CODE REVIEW

File: src/libsyntax/config.rs [hunk 1]
```diff
@@ -365,7 +353,3 @@ impl<'a> MutVisitor for StripUnconfigured<'a> {
         noop_visit_pat(pat, self)
     }
 }
-
-fn is_cfg(attr: &ast::Attribute) -> bool {
```
Comment (Centril) [inline src/libsyntax/config.rs]: Why this change?

**Comment by Centril** [comment]
I don't know whether we have any lint check attributes that affect generic parameters, but how does the language interact with those before / after this PR?

**Comment by Centril** [comment]
I see no reason not to do this and every reason to both in terms of consistency and usability. That said, since it is a language change, let's formally approve this change...

@rfcbot merge

**Comment by Centril** [comment]
N.B. Aaron is on leave so I've checked their box.

File: src/libsyntax/config.rs (hunk 1 above)
Comment (petrochenkov) [reply src/libsyntax/config.rs]: Drive-by cleanup.

**Comment by petrochenkov** [comment]
@Centril 
>I don't know whether we have any lint check attributes that affect generic parameters, but how does the language interact with those before / after this PR?

This change is orthogonal to lint checking attributes.
`allow`/`deny` and friends still don't work on generic parameters (which is a bug, but low priority).
```rust
fn f<#[allow(warnings)] foo>() {} // warning: type parameter `foo` should have an upper camel case name
```

File: src/libsyntax/config.rs (hunk 1 above)
Comment (Centril) [reply src/libsyntax/config.rs]: Yeah but I mean... it doesn't seem like cleanup but rather the opposite? Why not offer this function as an abstraction?

**Comment by Centril** [comment]
> This change is orthogonal to lint checking attributes.

Thanks! Would you mind filing an issue if there isn't one already?

File: src/libsyntax/config.rs (hunk 1 above)
Comment (petrochenkov) [reply src/libsyntax/config.rs]: Ok, will revert.

**Comment by petrochenkov** [comment]
>Would you mind filing an issue if there isn't one already?

https://github.com/rust-lang/rust/issues/61552

**Comment by Centril** [comment]
r? @Centril 

r=me rollup on the implementation when FCP completes.
